DOI.js

```diff
@@ -124,70 +121,50 @@ function completeDOIs(doc) {
 	}
 }
 
-function retrieveDOIs(dois, doc, providers) {
-	var numDois = dois.length;
-	var provider = providers.shift();
-	
-	var remainingDOIs = dois.slice();// copy array but not by reference
+function retrieveDOIs(dois, doc) {
+	let numDois = dois.length;
 
-	for (var i = 0, n = dois.length; i < n; i++) {
-		(function (doc, DOI) {
-			var translate = Zotero.loadTranslator("search");
-			translate.setTranslator(provider.id);
+	for (const DOI of dois) {
+		const translate = Zotero.loadTranslator("search");
+		translate.setTranslator("b28d0d42-8549-4c6d-83fc-8382874a5cb9");
 	
-			var item = { itemType: "journalArticle", DOI: DOI };
-			translate.setSearch(item);
+		translate.setSearch({ itemType: "journalArticle", DOI: DOI });
 	
-			// don't save when item is done
-			translate.setHandler("itemDone", function (translate, item) {
-				selectArray[item.DOI] = item.title;
-				if (!item.title) {
-					Zotero.debug("No title available for " + item.DOI);
-					item.title = "[No Title]";
-					selectArray[item.DOI] = "[" + item.DOI + "]";
-				}
-				items[item.DOI] = item;
-
-				// done means not remaining anymore
-				if (remainingDOIs.includes(item.DOI))	 {
-					remainingDOIs.splice(remainingDOIs.indexOf(item.DOI), 1);
-				}
-				else {
-					Z.debug(item.DOI + " not anymore in the list of remainingDOIs = " + remainingDOIs);
-				}
-			});
+		// don't save when item is done
+		translate.setHandler("itemDone", function (_translate, item) {
+			selectArray[item.DOI] = item.title;
+			if (!item.title) {
+				Zotero.debug("No title available for " + item.DOI);
+				item.title = "[No Title]";
+				selectArray[item.DOI] = "[" + item.DOI + "]";
+			}
+			items[item.DOI] = item;
+		});
 	
-			translate.setHandler("done", function (translate) {
-				numDois--;
-				if (numDois <= 0) {
-					Z.debug("Done with " + provider.name + ". Remaining DOIs: " + remainingDOIs);
-					if (providers.length > 0 && remainingDOIs.length > 0) {
-						retrieveDOIs(remainingDOIs, doc, providers);
-					}
-					else {
-						completeDOIs(doc);
-					}
-				}
-			});
+		translate.setHandler("done", function () {
+			numDois--;
+			/*
+				DOI resolution runs async, but updates of numDois run on the main thread, so if we are here that means:
```

Comment:
Everything still runs on the main thread, so this isn't really accurate, and it's also sort of misstating what's happening. Nothing is happening synchronously here other than kicking off the asynchronous translation process. As each process finishes, 'done' gets called and `numDois` is decremented. When the last async process is done, this will be 0. This is a pretty standard pattern, so I don't think we particularly need this comment.